### PR TITLE
[24.1] Add dynamic table log writer

### PR DESCRIPTION
**Pull request by achulkov2**
Adds a new `dynamic_table` log writer, which writes log events to the configured sorted or ordered dynamic table. Log events are serialized into a YSON representation, which is interpreted similarly to the `insert-rows` cli command. It is your responsibility to create the table with a compatible schema. See integration test for example configuration.

The writer is available for use in all YT server components.

---
bfb7bf098156a8b16a6fdbf018c54858b1479b03

Pull Request resolved: https://github.com/ytsaurus/ytsaurus/pull/631

### CODE REVIEW

File: yt/yt/tests/integration/misc/test_logging.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    DELTA_MASTER_CONFIG = {
+        "logging": {
+            "flush_period": 100,
+            "shutdown_busy_timeout": 1000,
+            "rules": [
+                {
+                    "family": "plain_text",
+                    "min_level": "debug",
+                    "exclude_categories": ["Bus"],
```
Comment (eshcherbin) [inline yt/yt/tests/integration/misc/test_logging.py:26]: We need the same fix here that was made for the trunk branch.

File: yt/yt/core/logging/unittests/logging_ut.cpp [hunk 2]
```diff
@@ -657,13 +666,14 @@ TEST_F(TLoggingTest, StructuredLoggingJsonFormat)
...
         /*commonFields*/ THashMap<TString, INodePtr>{},
-        /*enableControlMessages*/ true,
         /*enableSourceLocation*/ false,
         /*enableSystemFields*/ true,
+        /*enableHostField*/ false,
         jsonFormat);
 
     auto writer = CreateFileLogWriter(
         std::move(formatter),
+        CreateDefaultSystemLogEventProvider(writerConfig),
```
Comment (eshcherbin) [inline yt/yt/core/logging/unittests/logging_ut.cpp:676]: Please also cherry-pick this fix: https://github.com/ytsaurus/ytsaurus/commit/13d87d4aa9b78d1ac94d0e0d00633b4d40b5ae05

**Review by eshcherbin — APPROVED** [review]

File: yt/yt/tests/integration/misc/test_logging.py (hunk 1 above)
Comment (achulkov2) [reply yt/yt/tests/integration/misc/test_logging.py]: Done.

File: yt/yt/core/logging/unittests/logging_ut.cpp (hunk 2 above)
Comment (achulkov2) [reply yt/yt/core/logging/unittests/logging_ut.cpp:677]: And done.

**Comment by eshcherbin** [comment]
Shipped!

**Pull request closed without merging**